modules/tinymce/src/themes/silver/main/ts/ui/core/LineHeight.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import { Menu } from '@ephox/bridge';
+import { Arr, Optional, Unique } from '@ephox/katamari';
+import Editor from 'tinymce/core/api/Editor';
+
+const defaultLineHeights = '1 1.1 1.2 1.3 1.4 1.5 2';
+
+const getLineHeights = (editor: Editor): Menu.ToggleMenuItemSpec[] => {
+  const raw = editor.queryCommandValue('LineHeight');
+  const current = raw !== '' ? Optional.some(raw) : Optional.none<string>();
+  const alwaysAvailable = editor.getParam('lineheight_formats', defaultLineHeights, 'string').split(' ');
+
+  const allOptions = [ ...alwaysAvailable, ...current.toArray() ];
+  const deduped = Unique.stringArray(allOptions);
+  const sorted = Arr.sort(deduped, (x, y) => parseFloat(x) - parseFloat(y));
+
+  return Arr.map(
+    sorted,
+    (value) => ({
+      type: 'togglemenuitem',
+      active: current.is(value),
```

Comment:
This isn't consistent with how the rest of these buttons are done and misses a minor edge case (it doesn't update if the selection changes while the menu is open). So instead we should probably use the `onSetup: onSetupFormatToggle(editor, name)` logic that's used elsewhere.

Reply:
I'd hoped I could avoid that issue as people wouldn't be able to change focus while the menu is open. I'm guessing this is necessary in case someone changes selection with the API?

Reply:
Yeah, that's correct.

Reply:
I should add we've discussed deprecating the `active` property, as it's not really a good fit and leads to bugs. Ideally, any state update should be done in the `onSetup` function.